File: MATTHEW/leetcode/AIcoding/4.GPT-4o/go_solutions/1115-print-foobar-alternately.go
type FooBar struct {
    n int
    ch1 chan struct{}
    ch2 chan struct{}
}

func Constructor(n int) FooBar {
    return FooBar{n: n, ch1: make(chan struct{}), ch2: make(chan struct{})}
}

func (this *FooBar) Foo(printFoo func()) {
    for i := 0; i < this.n; i++ {
        printFoo()
        this.ch1 <- struct{}{}
        <-this.ch2
    }
}

func (this *FooBar) Bar(printBar func()) {
    for i := 0; i < this.n; i++ {
        <-this.ch1
        printBar()
        this.ch2 <- struct{}{}
    }
}

File: leetcode/AIcoding/5.Qwen-3/javascript_solutions/1115-print-foobar-alternately.js
var FooBar = function(n) {
    this.n = n;
};

FooBar.prototype.foo = function(printFoo) {
    for (let i = 0; i < this.n; i++) {
        printFoo();
    }
};

FooBar.prototype.bar = function(printBar) {
    for (let i = 0; i < this.n; i++) {
        printBar();
    }
};